### PR TITLE
Support GitHub actions

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Added `libs.versions.toml` support
 * Added `diktat` and `detekt` gradle plugins
 * Added `buildSrc` module for `diktat` and `detekt` configuration
 * Added GitHub hooks for linters and test checks

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      - name: Retrieve Kotlin version
+        run: |
+          kv=$(cat gradle/libs.versions.toml | grep '^kotlin =' | awk -F'[=]' '{print $2}' | tr -d '" ')
+          echo KOTLIN_VERSION=$kv >> $GITHUB_ENV
+      - name: Cache konan
+        uses: actions/cache@v3
+        with:
+          path: ~/.konan
+          key: ${{ runner.os }}-gradle-konan-${{ env.KOTLIN_VERSION }}
+      - name: Build all (excluding save-orchestrator-common, save-orchestrator and save-sandbox)
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply .github/workflows/build_and_test.yml:40]: oh, forgot to change the name, thx

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+[versions]
+kotlin = "1.8.20"
+kotlinx-datetime = "0.4.0"
+kotlinx-coroutines = "1.6.3-native-mt"
```
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply gradle/libs.versions.toml:4]: Sure, will do it

[the remark this answers is not in the document]
Comment (sanyavertolet) [reply gradle/libs.versions.toml]: Fixed

File: .github/workflows/build_and_test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply .github/workflows/build_and_test.yml]: Fixed

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+okio = "3.3.0"
+# linters
+diktat = "1.2.5"
+detekt = "1.22.0"
+# logging
+kotlin-logging = "4.0.0-beta-28"
```
[the remark this answers is not in the document]
Comment (nulls) [reply gradle/libs.versions.toml:10]: yeap, 4.x is not backward compatible. for reference: https://github.com/oshai/kotlin-logging/issues/264

File: build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+
+import com.saveourtool.processbuilder.configureDetekt
+import com.saveourtool.processbuilder.configureDiktat
+import com.saveourtool.processbuilder.createDetektTask
+
+group = "com.saveourtool"
+description = "Multiplatform ProcessBuilder implementation"
```
Comment (nulls) [inline build.gradle.kts:7]: just a not to naming: maybe just `Kotlin Process Builder`?

File: buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DiktatConfiguration.kt [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            exclude("build/**")
+        }
+    }
+    fixDiktatTask()
+}
+
+private fun Project.fixDiktatTask() {
+    tasks.withType<DiktatJavaExecTaskBase>().configureEach {
+        // https://github.com/saveourtool/diktat/issues/1269
+        systemProperty("user.home", rootDir.toString())
```
Comment (nulls) [inline buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DiktatConfiguration.kt:34]: is this issue still relevant?

File: diktat-analysis.yml [hunk 6]
```diff
@@ -0,0 +1,114 @@
+- name: DIKTAT_COMMON
```
Comment (nulls) [inline diktat-analysis.yml:1]: why do we have so many overrides for rules in our own project?
I understand the configuration for domain name and srcDirectory, but we override the rest?

**Review by nulls — APPROVED** [review]

File: buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DiktatConfiguration.kt (hunk 5 above)
Comment (sanyavertolet) [reply buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DiktatConfiguration.kt:34]: Seems like no, will remove `fixDiktatTask`

File: build.gradle.kts (hunk 4 above)
Comment (sanyavertolet) [reply build.gradle.kts:7]: Ok, will do it

File: diktat-analysis.yml (hunk 6 above)
Comment (sanyavertolet) [reply diktat-analysis.yml:1]: Removed rule configuration overrides that actually do not override anything

File: build.gradle.kts (hunk 4 above)
Comment (sanyavertolet) [reply build.gradle.kts]: Done

File: buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DiktatConfiguration.kt (hunk 5 above)
Comment (sanyavertolet) [reply buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DiktatConfiguration.kt]: Done

File: buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DetektConfiguration.kt [hunk 7]
```diff
@@ -19,32 +19,35 @@ fun Project.configureDetekt() {
...
         }
     }
 
     tasks.register("mergeDetektReports", ReportMergeTask::class) {
+        mustRunAfter(detektAllTask)
         output.set(buildDir.resolve("detekt-sarif-reports/detekt-merged.sarif"))
     }
+
+    @Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
+    val reportMerge: TaskProvider<ReportMergeTask> = rootProject.tasks.named<ReportMergeTask>("mergeDetektReports") {
```
Comment (nulls) [inline buildSrc/src/main/kotlin/com/saveourtool/processbuilder/DetektConfiguration.kt:43]: it's the same task as created on prev line `tasks.register("mergeDetektReports", ReportMergeTask::class) {`
why not merged them?

**Pull request merged**